apollo-ios-codegen/Sources/ApolloCodegenLib/CodegenConfiguration/ApolloCodegenConfiguration+RenameSchemaTypesConfiguration.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+
+struct SchemaCustomization: Codable {
+  
+  let customTypeNames: [String: CustomSchemaTypeName]
+ 
+  // MARK: - Codable
+  
+  enum CodingKeys: CodingKey, CaseIterable {
+    case customTypeNames
+  }
+  
+  public init(from decoder: Decoder) throws {
+    let values = try decoder.container(keyedBy: CodingKeys.self)
+    try throwIfContainsUnexpectedKey(
+      container: values,
+      type: Self.self,
+      decoder: decoder
+    )
+    
+    customTypeNames = try values.decode(
+      [String: CustomSchemaTypeName].self,
+      forKey: .customTypeNames
+    )
+  }
+  
+  public func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    
+    try container.encode(self.customTypeNames, forKey: .customTypeNames)
+  }
+  
+  // MARK: - Enums
+  
+  enum CustomSchemaTypeName: Codable {
+    case object(name: String)
+    case interface(name: String)
+    case customScalar(name: String)
+    case union(name: String)
+    case `enum`(name: String, cases: [String: String])
```

Comment:
I think we might want to make `name` an optional `String?` on this and `inputObject`. You may want to change the name of cases/fields without changing the name of the type itself.

Reply:
I agree ease of use is important, but I also think in general some extra complexity for more explicit clarity of what is happening can be useful as well, in this case both from the JSON side and especially having the different enum cases on the swift side in case there may be times that information is needed. I think a setup like that is a nice middle ground between just strings as you suggested, and having a directive which would make it very clear what type was being modified. That said, I'm happy to make the suggested changes and focus on ease of use.

Also yea the `name` will need to be optional for the scenario you described.